### ISSUE TITLE
x509: certificate is valid for docker internal address, not advirtisedAddress

### ISSUE THREAD

**Issue by albertpx** [kumahq/kuma#6498]
### What happened?

We have configured a universal multi-zone cluster.

Our services are running on instances with docker. Each instance has a single service running as a docker container, and we put the dataplane as another container in the same docker bridge network.

The following configuration is used for dataplanes (an example of inbound only dataplane):
```yaml
type: Dataplane
mesh: default
name: {{ name }}
networking:
  address: {{ container_address }}
  advertisedAddress: {{ instance_address }}
  inbound:
    - port: 5555
      servicePort: 6666
      serviceAddress: 172.18.1.10
      tags:
        kuma.io/service: kuma-test-inbound
        kuma.io/protocol: http
```
When trying to run the following command:
```
kumactl inspect dataplane kuma-test-instance --type=config-dump
```
An `Internal Server Error` is returned, and inside a global control plane instance, we see the following in the log:
```
ERROR   Could not execute admin operation
{"error": "error response from Zone CP: unable to send GET to config_dump: Get \"https://10.160.0.115:9901/config_dump\": x509: certificate is valid for 172.18.0.2, not 10.160.0.115", 
"errorVerbose": "error response from Zone CP: unable to send GET to config_dump: Get \"https://10.160.0.115:9901/config_dump\": x509: certificate is valid for 172.18.0.2, not 10.160.0.115\ngithub.com/kumahq/kuma/pkg/envoy/admin.(*kdsEnvoyAdminClient).ConfigDump\n\t/home/circleci/project/pkg/envoy/admin/kds_client.go:66\ngithub.com/kumahq/kuma/pkg/api-server.inspectDataplaneAdmin.func1\n\t/home/circleci/project/pkg/api-server/inspect_envoy_admin_endpoints.go:117\ngithub.com/emicklei/go-restful/v3.(*FilterChain).ProcessFilter\n\t/home/circleci/go/pkg/mod/github.com/emicklei/go-restful/v3@v3.9.0/filter.go:23\ngithub.com/emicklei/go-restful/v3.CrossOriginResourceSharing.Filter\n\t/home/circleci/go/pkg/mod/github.com/emicklei/go-restful/v3@v3.9.0/cors_filter.go:53\ngithub.com/emicklei/go-restful/v3.(*FilterChain).ProcessFilter\n\t/home/circleci/go/pkg/mod/github.com/emicklei/go-restful/v3@v3.9.0/filter.go:21\ngithub.com/kumahq/kuma/pkg/plugins/authn/api-server/tokens.UserTokenAuthenticator.func1\n\t/home/circleci/project/pkg/plugins/authn/api-server/tokens/authenticator.go:34\ngithub.com/emicklei/go-restful/v3.(*FilterChain).ProcessFilter\n\t/home/circleci/go/pkg/mod/github.com/emicklei/go-restful/v3@v3.9.0/filter.go:21\ngithub.com/kumahq/kuma/pkg/api-server/authn.LocalhostAuthenticator\n\t/home/circleci/project/pkg/api-server/authn/localhost.go:25\ngithub.com/emicklei/go-restful/v3.(*FilterChain).ProcessFilter\n\t/home/circleci/go/pkg/mod/github.com/emicklei/go-restful/v3@v3.9.0/filter.go:21\ngithub.com/kumahq/kuma/pkg/util/prometheus.MetricsHandler.func1.1\n\t/home/circleci/project/pkg/util/prometheus/gorestful_middleware.go:21\ngithub.com/slok/go-http-metrics/middleware.Middleware.Measure\n\t/home/circleci/go/pkg/mod/github.com/slok/go-http-metrics@v0.10.0/middleware/middleware.go:117\ngithub.com/kumahq/kuma/pkg/util/prometheus.MetricsHandler.func1\n\t/home/circleci/project/pkg/util/prometheus/gorestful_middleware.go:20\ngithub.com/emicklei/go-restful/v3.(*FilterChain).ProcessFilter\n\t/home/circleci/go/pkg/mod/github.com/emicklei/go-restful/v3@v3.9.0/filter.go:21\ngithub.com/emicklei/go-restful/v3.(*Container).dispatch\n\t/home/circleci/go/pkg/mod/github.com/emicklei/go-restful/v3@v3.9.0/container.go:296\nnet/http.HandlerFunc.ServeHTTP\n\t/home/circleci/go/src/net/http/server.go:2084\nnet/http.(*ServeMux).ServeHTTP\n\t/home/circleci/go/src/net/http/server.go:2462\nnet/http.serverHandler.ServeHTTP\n\t/home/circleci/go/src/net/http/server.go:2916\nnet/http.(*conn).serve\n\t/home/circleci/go/src/net/http/server.go:1966\nruntime.goexit\n\t/home/circleci/go/src/runtime/asm_amd64.s:1571"}
```

It appears that the envoy admin api certificate is generated only for the internal docker bridge address, but not for the advertisedAddress (the ip address of the instance).

According to the docs here: https://kuma.io/docs/2.1.x/generated/resources/proxy_dataplane/#dataplane
The address of admin api will always listen on `address` and not on `advertisedAddress` - which I assume is fine, but then why is the control plane trying to communicate using the `advertisedAddress`? And if that is on purpose, then shouldn't the certificate include the `advertisedAddress`?

**Comment by lobkovilya** [comment]
Here https://github.com/kumahq/kuma/blob/master/pkg/xds/sync/dataplane_watchdog.go#L121, we have to generate certificates both for Address and AdvertisedAddress (same for Ingress/Egress)

**Comment by dascole** [comment]
I'll grab this one

**Comment by mmorel-35** [comment]
This issue shall be closed as #6584 is closed now
